packages/block-editor/src/components/link-control/style.scss

```diff
@@ -137,39 +111,24 @@ $preview-image-height: 140px;
 }
 
 .block-editor-link-control__search-item {
-	position: relative;
-	display: flex;
-	align-items: flex-start; // when link text is very long it is important this indicator remains visible and thus should be aligned top.
-	font-size: $default-font-size;
-	cursor: pointer;
-	background: $white;
-	width: 100%;
-	border: none;
-	text-align: left;
-	padding: $grid-unit-15 $grid-unit-20;
-	border-radius: 2px;
-	height: auto;
-
-	&:hover,
-	&:focus {
-		background-color: $gray-100;
 
-		.block-editor-link-control__search-item-type {
-			background: $white;
-		}
+	&.components-button.components-menu-item__button {
+		height: auto;
+		text-align: left;
 	}
 
-	// The added specificity is needed to override.
-	&:focus:not(:disabled) {
-		box-shadow: 0 0 0 var(--wp-admin-border-width-focus) var(--wp-admin-theme-color) inset;
+	.components-menu-item__item {
+		overflow: hidden;
+		text-overflow: ellipsis;
+		// Inline block required to preserve white space
+		// between `<mark>` elements and text nodes.
+		display: inline-block;
 	}
 
-	&.is-selected {
-		background: $gray-100;
-
-		.block-editor-link-control__search-item-type {
-			background: $white;
-		}
```

Comment:
Removing this completely removed all "selected" styling from the list. I restored it in https://github.com/WordPress/gutenberg/pull/50978/commits/cecd7c8ba190c768f1717287cb8b6f04c0885f54